scripts/create_dataset.py

```diff
@@ -46,26 +46,323 @@ def init_dataset(dataset_name, dataset_description):
+        if len(relationships) > 0:
+            dataset["schema"]["relationships"].extend(relationships)
+
+    return dataset
+
+
+def create_and_add_assets(input_assets, dataset):
+    """
+    For every input asset it creates an asset by adding all the tables and relationship names to the asset and
+    then adds each created assets to the dataset.
+    :param input_assets: the input json assets that have been added to a dict
+    :param dataset: the created dataset that will be written to disk
+    """
+    for input_asset_name in input_assets:
+        asset = create_asset(input_asset=input_assets[input_asset_name],
+                             dataset=dataset)
+        dataset["schema"]["assets"].append(asset)
+
     return dataset
 
 
+def create_asset(input_asset, dataset):
+    """
+    Creates an asset by adding all the tables and relationship names to the asset.
+    :param input_asset: the input json asset that has been added to a dict
+    :param dataset: the created dataset that will be written to disk
+    """
+    asset = init_asset(asset_name=input_asset["name"],
+                       root_table=input_asset["rootTable"],
+                       root_column=input_asset["rootColumn"])
+
+    # add tables and the follow section to the asset
+    asset_with_tables = add_asset_tables(asset=asset, dataset=dataset)
+    asset_with_tables_and_follow = add_asset_follow(asset=asset_with_tables, dataset=dataset)
+
+    return asset_with_tables_and_follow
+
+
+def add_asset_tables(asset, dataset):
+    """
+    Adds all the table names from the dataset to the asset.
+    :param asset: the asset to be created and added to the dataset with table names
+    :param dataset: the created dataset that will be written to disk
+    """
+    for table in dataset["schema"]["tables"]:
+        asset["tables"].append(
+            {
+                "name": table["name"],
+                "columns": []
+            }
+        )
+    return asset
+
+
+def add_asset_follow(asset, dataset):
+    """
+    Adds all the table names from the dataset to the asset.
+    :param asset: the asset to be created and added to the dataset with relationship names
+    :param dataset: the created dataset that will be written to disk
+    """
+    for relationship in dataset["schema"]["relationships"]:
+        asset["follow"].append(relationship["name"])
+
+    return asset
+
+
+def init_asset(asset_name, root_table, root_column):
+    """
+    Adds all the table names from the dataset to the asset.
+    :param asset_name: the name of the asset being  created and added to the dataset
+    :param root_table: the name of the root table for the asset
+    :param root_column: the name of the root column for the asset
+    """
+    asset = {
+        "name": asset_name,
+        "rootTable": root_table,
+        "rootColumn": root_column,
+        "tables": [],
+        "follow": []
+    }
+    return asset
```

Comment:
Busting out a method for this feels like overkill. It could be written as:
```python
return {
    "name": asset_name,
    "rootTable": root_table,
    "rootColumn": root_column,
    "tables": [],
    "follow": []
}
```
But then there's no real reason not to inline it at the current callsite.